File: TDDObjectiveC/Models/Euro.m
//
//  Euro.m
//  TDDObjectiveC
//
//  Created by Alan Casas on 26/10/2019.
//  Copyright © 2019 Alan Casas. All rights reserved.
//

#import "Euro.h"

@interface Euro()
@property (nonatomic) NSInteger amount;

@end

@implementation Euro

-(id) initWithAmount: (NSInteger) amount {
    
    if (self = [super init]) {
        _amount = amount;
    }
    
    return self;
    
}

-(Euro *) times:(NSInteger) multiplier {
    Euro *newEuro = [[Euro alloc] initWithAmount:self.amount * multiplier];
    return newEuro;
}

@end

File: TDDObjectiveC/Models/Euro.swift
//
//  Euro.swift
//  TDDSwift
//
//  Created by Alan Casas on 26/10/2019.
//  Copyright © 2019 Alan Casas. All rights reserved.
//

import Foundation

class Euro {
    
    var amount: Int!
    
    init(initWithAmount amount: Int) {
        self.amount = amount
    }
    
    func times (multiplier times: Int) -> Euro {
        let newEuro = Euro(initWithAmount: self.amount * times)
        return newEuro
    }

}

extension Euro: Equatable {
    static func == (lhs: Euro, rhs: Euro) -> Bool {
        if lhs == rhs {
            return true
        } else {
            return false
        }        
    }
}
